megatron/training.py

```diff
@@ -749,8 +750,37 @@ def train(forward_step_func, model, optimizer, lr_scheduler,
     timers('interval-time').start()
     print_datetime('before the start of training step')
     report_memory_flag = True
+
+    # flush intervals prior to current iteration
+    if args.skip_train_iteration_range is not None:
+        ends = [end for start, end in args.skip_train_iteration_range]
+        index = bisect.bisect_left(ends, iteration)
+        for _ in range(index):
+            args.skip_train_iteration_range.popleft()
+
     while iteration < args.train_iters and (args.train_tokens is None or \
         args.consumed_train_tokens < args.train_tokens):
+        if (
+            train_data_iterator is not None
+            and args.skip_train_iteration_range
```

Comment:
```suggestion
            and args.skip_train_iteration_range is not None
            and len(args.skip_train_iteration_range) > 0
```

It helps when the check is well defined. I suppose that this is what you meant. Please adjust if you wanted a different type of check

Reply:
This is exactly what I meant, thank you!